### ISSUE TITLE
Allow the "Using subdirectories" method of creating paged content include OCR and HOCR files

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#639]
On Slack, borwickja asked if the "Using subdirectories" method of creating paged content allowed inclusion of OCR and HOCR files along with the page images. Currently, no, but that's a great idea! 

Currently, the following directory structure will create two book nodes with the attached paged content members:

```
books/
├── book1
│   ├── page-001.jpg
│   ├── page-002.jpg
│   └── page-003.jpg
├── book2
│   ├── isbn-1843341778-001.jpg
│   ├── using-islandora-workbench-page-002.jpg
│   └── page-003.jpg
└── metadata.csv
```

It would be useful to be able to include pre-generated OCR and HOCR files in these directories like this:

```
books/
├── book1
│   ├── page-001.jpg
│   ├── page-001.ocr
│   ├── page-001.hocr
│   ├── page-002.jpg
│   ├── page-002.ocr
│   ├── page-002.hocr
│   ├── page-003.jpg
│   ├── page-003.ocr
│   └── page-003.hocr
├── book2
│   ├── isbn-1843341778-001.jpg
│   ├── isbn-1843341778-001.ocr
│   ├── isbn-1843341778-001.hocr
│   ├── using-islandora-workbench-page-002.jpg
│   ├── using-islandora-workbench-page-002.ocr
│   ├── using-islandora-workbench-page-002.hocr
│   ├── page-003.jpg
│   ├── page-003.ocr
│   └── page-003.hocr
└── metadata.csv
```

We'd need to create a configuration setting to map file extensions to Media Use terms and do some other things to make sure the OCR and HOCR files got ingested as required.

**Comment by mjordan** [comment]
[Conversation on Slack](https://islandora.slack.com/archives/CM5PPAV28/p1704737845127919) indicates that we haven't settled on a single way to managing hOCR files - the two approach are 1) as separate media identified by a term from the Media Use vocabulary and 2) attached to another media a la .vtt track file. 

It probably makes sense to prepare Workbench to handle both approaches.

**Comment by mjordan** [comment]
Docs will need to indicate that users should deselect the "Extract text from image or PDF" action in the "Page derivatives" Context before using this method. If this will require a new config setting (probably should, to map file extensions in the directory with Media Use terms), Workbench could tell the user during `--check` to temporarily deselect this action.

**Comment by mjordan** [comment]
Since this task already requires `paged_content_from_directories: true`, I'd like to propose that we add the following setting to map Media use terms to extensions:

```
paged_content_from_directories_media:
 - Extracted text: txt
 - hOCR: xml
 ```
(note that I include `- hOCR: xml` only to illustrate a second mapping). We can also allow Media Use term URIs ro term IDs in the left-hand part of the mapping.

**Comment by mjordan** [comment]
`add_media` tasks already emit warnings during `--check` like `Warning: Term ID "36" provided in configuration option "media_use_tid" will assign an Islandora Media Use term that might conflict with derivative media.  You should temporarily disable the Context or Action that generates those derivatives.`

**Comment by mjordan** [comment]
Resolved with eea1165a888769be87246dc15352446023fbb612. New docs are at https://mjordan.github.io/islandora_workbench_docs/paged_and_compound/#ingesting-ocr-and-other-files-with-page-images.
